c++/src/kj/string.h

```diff
@@ -721,15 +721,13 @@ inline constexpr ArrayPtr<const char> StringPtr::asArray() const {
 }
 
 inline bool StringPtr::operator==(const StringPtr& other) const {
-  return content.size() == other.content.size() &&
-      memcmp(content.begin(), other.content.begin(), content.size() - 1) == 0;
+  // The -1 is to ignore the NUL character
+  return content.first(content.size() - 1) == other.content.first(other.content.size() - 1);
```

Comment:
Kind of wonder if it's better to just do `content == other.content` here, the -1s might cost more time than they save?

Reply:
Should I try to measure this?

Reply:
I think you should just change it and not worry about measuring. The speed difference is likely negligible but it's nice to reduce code complexity.

Reply:
Especially given that `operator<` doesn't use the -1. They might as well be consistent.